### PR TITLE
[SPARK-45665][INFRA] Uses different `ORACLE_DOCKER_IMAGE_NAME` in the scheduled builds in other branches

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
https://github.com/apache/spark/pull/43123 upgraded the version of Oracle used for testing. As the daily test will reuse `build_and_test.yml`, the branch-3.x will also use the new version. However, due to the lack of synchronized code changes, the `OracleIntegrationSuite` in the `Docker integration tests` cannot pass the test during the daily test of branch-3.x:

- branch-3.3: https://github.com/apache/spark/actions/runs/6609791712/job/17950549755
- branch-3.4: https://github.com/apache/spark/actions/runs/6611049884/job/17954225189
- branch-3.5: https://github.com/apache/spark/actions/runs/6612344747/job/17958021656

So this PR adds the ORACLE_DOCKER_IMAGE_NAME environment variable to the daily test yml file for branch-3.x and uses the previous version of the Oracle Docker image that can pass the test successfully.

### Why are the changes needed?
Restore the daily test for branch-3.x.

### Does this PR introduce _any_ user-facing change?
No,dev-only

### How was this patch tested?
Monitor the daily test results.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: .github/workflows/build_branch33.yml [hunk 1]
```diff
@@ -37,7 +37,8 @@ jobs:
       envs: >-
         {
           "SCALA_PROFILE": "scala2.13",
-          "PYTHON_TO_TEST": ""
+          "PYTHON_TO_TEST": "",
+          "ORACLE_DOCKER_IMAGE_NAME": "gvenzl/oracle-xe:21.3.0"
```
Comment (LuciferYang) [inline .github/workflows/build_branch33.yml:41]: The `build_and_test.yml` in branch-3.3 is configured as `oracle-xe:18.4.0`, but the previous daily test has always used `oracle-xe:21.3.0`, so it is still configured as `oracle-xe:21.3.0` here.

File: .github/workflows/build_branch34.yml [hunk 2]
```diff
@@ -37,7 +37,8 @@ jobs:
       envs: >-
         {
           "SCALA_PROFILE": "scala2.13",
-          "PYTHON_TO_TEST": ""
+          "PYTHON_TO_TEST": "",
+          "ORACLE_DOCKER_IMAGE_NAME": "gvenzl/oracle-xe:21.3.0"
```
Comment (LuciferYang) [inline .github/workflows/build_branch34.yml:41]: I tested it in my own repository, the environment variable did not overwrite successfully, and I haven't found the reason yet.

**Comment by LuciferYang** [comment]
I've tested this solution in my personal repository, and it should be feasible.

branch-3.3: https://github.com/LuciferYang/spark/actions/runs/6643102365/job/18054383056

image


branch-3.4: https://github.com/LuciferYang/spark/actions/runs/6637023118

image


branch-3.5: https://github.com/LuciferYang/spark/actions/runs/6637015146/job/18065960793

image

**Comment by LuciferYang** [comment]
cc @HyukjinKwon @dongjoon-hyun

**Review by HyukjinKwon — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged into master. I will keep an eye on tomorrow's daily test.

**Comment by LuciferYang** [comment]
Thanks @HyukjinKwon ~

**Comment by dongjoon-hyun** [comment]
+1, late LGTM.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun ~

**Comment by LuciferYang** [comment]
branch-3.3:https://github.com/apache/spark/actions/runs/6650581612
branch-3.4: https://github.com/apache/spark/actions/runs/6651853463
branch-3.5:https://github.com/apache/spark/actions/runs/6653188893

All passed ~

**Pull request closed without merging**